src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -377,9 +377,10 @@ private static void ParseRange(string source, ref int offset, ref RangeGroup ran
             }
         }
 
-        public static TimeValue ParseLastFromString(string source)
+        public static (TimeSpan TimePeriod, TimeValue Months) ParseTimePeriodFromString(string source)
```

Comment:
Why can't we use TimeValue?

Reply:
`TimeValue` has 2 native units `Seconds` & `Months`
We can either change from `Seconds` to `MilliSeconds`, but this will break compatibility and limit the max expressed value to 60 days (it is also an `int`)
Adding an additional unit of `Milliseconds` will complicate the code, so we thought that adding `TimeSpan` here is the best solution.